storage/src/main/java/com/strategyobject/substrateclient/storage/StorageKeyProvider.java

```diff
@@ -0,0 +1,126 @@
+package com.strategyobject.substrateclient.storage;
+
+import com.strategyobject.substrateclient.rpc.types.StorageKey;
+import lombok.*;
+import net.openhft.hashing.LongHashFunction;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class StorageKeyProvider {
+    private static final int XX_HASH_SIZE = 16;
+    @Getter(AccessLevel.PACKAGE)
+    private final List<KeyHasher<?>> keyHashers = new ArrayList<>();
+    private final ByteBuffer keyPrefix;
+
+    private StorageKeyProvider(String palletName, String storageName) {
+        keyPrefix = ByteBuffer.allocate(XX_HASH_SIZE * 2);
+
+        xxHash128(keyPrefix, palletName);
+        xxHash128(keyPrefix, storageName);
+    }
+
+    private static void xxHash128(ByteBuffer buf, String value) {
+        val encodedValue = value.getBytes(StandardCharsets.UTF_8);
+
+        final ByteOrder sourceOrder = buf.order(); // final ByteOrder instead of val because of checkstyle
+        buf.order(ByteOrder.LITTLE_ENDIAN);
+
+        buf.asLongBuffer()
+                .put(LongHashFunction.xx(0).hashBytes(encodedValue))
+                .put(LongHashFunction.xx(1).hashBytes(encodedValue));
+
+        buf.position(buf.position() + XX_HASH_SIZE);
+        buf.order(sourceOrder);
+    }
+
+    public static StorageKeyProvider with(@NonNull String palletName, @NonNull String storageName) {
```

Comment:
might be confusing to have two different methods with the same name. what do you think if this one is called `for`?
`StorageKeyProvider.for("Society", "Votes").with(KeyHasher....)`

Reply:
`for` is a key word, so it's not acceptable. Renamed to `of` and `use` properly.